packages/plugin-core/src/commands/MigrateSelfContainedVault.ts

```diff
@@ -0,0 +1,92 @@
+import { DendronError, DVault, VaultUtils } from "@dendronhq/common-all";
+import { WorkspaceService } from "@dendronhq/engine-server";
+import _ from "lodash";
+import * as vscode from "vscode";
+import { DENDRON_COMMANDS } from "../constants";
+import { IDendronExtension } from "../dendronExtensionInterface";
+import { ExtensionProvider } from "../ExtensionProvider";
+import { VSCodeUtils } from "../vsCodeUtils";
+import { BasicCommand } from "./base";
+
+type CommandOpts = {
+  /** Which vault to migrate? */
+  vault?: DVault;
+};
+type CommandOutput = {
+  /** The vault after the migration, or null if the migration was cancelled. */
+  newVault: DVault | null;
+};
+
+export class MigrateSelfContainedVaultCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  key = DENDRON_COMMANDS.MIGRATE_SELF_CONTAINED.key;
+  private extension: IDendronExtension;
+
+  async sanityCheck(opts?: CommandOpts): Promise<undefined | string> {
+    if (opts?.vault && VaultUtils.isSelfContained(opts.vault)) {
+      return "Already a self contained vault";
+    }
+    return undefined;
+  }
+
+  constructor(ext: IDendronExtension) {
+    super();
+    this.extension = ext;
+  }
+
+  async gatherInputs(opts?: CommandOpts): Promise<CommandOpts | undefined> {
```

Comment:
since this is a destructive operation, can we warn folks to backup all their notes before doing this?